.github/workflows/push.yml

```diff
@@ -0,0 +1,18 @@
+on: push
+name: Continuous Integration
```

Comment:
Can we break these sections apart, and put the `name` at the top?

Reply:
We might be able to, the creation of the image to run them in is the expensive part and as far as I can tell we will lose the caching of that extending the run time of these checks quite a bit.

Reply:
The new details view (1 click away) is a lot better in the new github actions, that might be good enough?

Reply:
I don't follow - I'm just looking to see if we can improve the readability of the yaml slightly:

```yaml
name: "Continuous Integration"

on: push

jobs:
  [...]
```

Reply:
done and done 👍